src/main/java/tech/aroma/data/sql/SQLUserPreferencesRepository.kt

```diff
@@ -0,0 +1,175 @@
+/*
+ * Copyright 2017 RedRoma, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package tech.aroma.data.sql
+
+import org.slf4j.LoggerFactory
+import org.springframework.dao.EmptyResultDataAccessException
+import org.springframework.jdbc.core.JdbcOperations
+import tech.aroma.data.UserPreferencesRepository
+import tech.aroma.data.assertions.RequestAssertions.validMobileDevice
+import tech.aroma.data.assertions.RequestAssertions.validUserId
+import tech.aroma.data.sql.SQLStatements.*
+import tech.aroma.thrift.channels.MobileDevice
+import tech.aroma.thrift.exceptions.InvalidArgumentException
+import tech.aroma.thrift.exceptions.OperationFailedException
+import tech.sirwellington.alchemy.arguments.Arguments.checkThat
+import tech.sirwellington.alchemy.thrift.ThriftObjects
+import javax.inject.Inject
+
+
+/**
+ *
+ * @author SirWellington
+ */
+internal class SQLUserPreferencesRepository
+@Inject constructor(val database: JdbcOperations, val serializer: DatabaseSerializer<MutableSet<MobileDevice>>) : UserPreferencesRepository
+{
+    override fun saveMobileDevice(userId: String, mobileDevice: MobileDevice)
+    {
+        checkUserId(userId)
+        checkMobileDevice(mobileDevice)
+
+        val sql = Inserts.USER_DEVICE
+        val serialized = ThriftObjects.toJson(mobileDevice)
```

Comment:
You still ensure the exception is mapped.
![bitmoji](https://render.bitstrips.com/v2/cpanel/10212615-288439948_4-s1-v1.png?transparent=1&palette=1&width=246)